python/pyspark/pandas/data_type_ops/boolean_ops.py

```diff
@@ -122,13 +123,23 @@ def floordiv(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
             raise TypeError(
                 "Floor division can not be applied to %s and the given type." % self.pretty_name
             )
+        spark_session = left._internal.spark_frame.sparkSession
+
+        def safe_floordiv(left_col: PySparkColumn, right_val: Any) -> PySparkColumn:
+            if is_ansi_mode_enabled(spark_session):
+                return F.when(F.lit(right_val == 0), F.lit(None)).otherwise(
```

Comment:
Can we just always use this expression? Checking the configuration is actually pretty expensive especially in Spark Connect.

Reply:
When ANSI is off we need to preserve the previous behavior, that's why we we check the configs here

Reply:
When ANSI is off, it just returns `None` which you're handling now .. With ANSI on / off, the behaviour here should be the same anyway.

Reply:
Good idea,used try_divide for code consistency